### ISSUE TITLE
DemoFile.cancel() not working

### ISSUE THREAD

**Issue by derpalmer** [saul/demofile#20]
As I try to build a workaround for #11 I get an error when trying to cancel the parsing.

```
D:\git\demoanalyzer\node_modules\demofile\demo.js:287
      timers.cancelImmediate(this._immediateTimerToken);
             ^

TypeError: timers.cancelImmediate is not a function
    at DemoFile.cancel (D:\git\demoanalyzer\node_modules\demofile\demo.js:287:14)
    at DemoFile.demo.on.v (D:\git\demoanalyzer\demoworker.js:70:14)
    at emitOne (events.js:96:13)
    at DemoFile.emit (events.js:188:7)
    at DemoFile._parseRecurse (D:\git\demoanalyzer\node_modules\demofile\demo.js:297:10)
    at runCallback (timers.js:666:20)
    at tryOnImmediate (timers.js:639:5)
    at processImmediate [as _immediateCallback] (timers.js:611:5)
```

As described in the timers documentation the function name is "clearImmediate" but even when I change it the parsing goes on. Is there anything else wrong?
https://nodejs.org/api/timers.html#timers_cancelling_timers

Node v6.10.2

**Comment by saul** [comment]
Looking at the call stack it looks like the problem is that after you cancel, the _parseRecurse sets up recursion again.

Can you try moving this._recurse() to the top of _parseRecurse (it's currently at the bottom) and let me know if that fixes it. If so I will release an update.

**Comment by derpalmer** [comment]
Works just fine, thanks! Created a PR #21

**Comment by saul** [comment]
Fixed and published in 8f65012a2127656dbfe404f69a68a3fc7660b2bd
